### ISSUE TITLE
Training from scratch - log report

### ISSUE THREAD

**Issue by moabarar** [YimianDai/open-aff#11]
Hi,

First, I think your paper is very interesting, excellent work!

I was wondering if you have training from scratch report avialable ?
All the aviable reports are based on pretrained models with already high-accuracy (specifically, I am referring to CIFAR100 expierement).

**Comment by YimianDai** [comment]
Hi, I have uploaded the training log from scratch with the prefix "From_Scratch". You can find them in the corresponding folders now. 

The ResNet result is the same, but the ResNeXt result is lower than training on pretrained models.

**Comment by moabarar** [comment]
Thanks a lot for the quick reply! 
Again - well done!

**Comment by sbl1996** [comment]
Can you provide the training script used for training from scratch ? It seems that the current provided training log from scratch does not use 'train_cifar.py' in this repo.

**Comment by YimianDai** [comment]
Sorry. Can you explain a bit more? I do not understand what "the current provided training log from scratch does not use 'train_cifar.py' in this repo." means?

What I have in mind is that I just renamed `train_cifar_mixup.py` to  `train_cifar.py` when I create this repo.  Everything else is the same. It is because my private repo is very redundant and has many codes unrelated with this paper. Therefore, I created this new public repo when I release the code.

**Comment by sbl1996** [comment]
1. As shown in `open-aff/params/cifar100/AFF-ResNet-32/From_Scratch_Log_train_cifar100_cifar100-ASKCFuse-resnet-32-c-4-s-1.log`, the log line is "INFO:root:[Epoch 0] train=0.079624 val=0.107400 loss=3.099419 time: 20.976282". However, in `train_cifar.py`, the log format is `[Epoch %d] train=%f val=%f loss=%f lr: %f time: %f`. They are different.

2. The training loss curve in the training log from scratch is shown as below.
![image](https://user-images.githubusercontent.com/13728074/97946292-f2d87680-1dc4-11eb-94b7-8f70a723ec04.png)
It seems that you have used `step` learning rate schedule. But in `train_cifar.py`, it is `cosine`.

**Comment by YimianDai** [comment]
In `train_cifar.py`, you can choose cosine or step, not cosine only.  It depends whether you add `--cosine` in the training log.

I start with step at first. When I used the pretrained model to try to have a higher accuracy I add the cosine code. But it seems that cosine didn't perform better than step on my cifar experiment. However, as long as the network is not changed, the network can use the stored params.

**Comment by sbl1996** [comment]
Thank you, I have noticed --cosine option. Except for printing lr, are there any other differences between the training script you used for training from scratch and `train_cifar.py`?

**Comment by YimianDai** [comment]
I think I only have made two changes on the code `train_cifar.py` when I started to run the very first experiment.  The first is adding the label smoothing choice, the second is the adding cosine choice.

What I have in mind is that the training log for `AFF-ResNeXt-38-32x4d` was not trained with label smoothing because it is trained before I added the label smoothing choice. But the rest models are all with label smoothing. 

Sorry for the inconvenience about the updates in my code. The ideal case is that I trained all these models after everything is fixed with the final version of code. But I have very limited access to GPUs, so I basically presented the training logs and params on the fly.
